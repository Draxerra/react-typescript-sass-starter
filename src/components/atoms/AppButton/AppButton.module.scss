.app-button {
  padding: 0.75em 1em;
  border: 0;
  border-radius: 0.5em;
  background-color: var(--clr-primary-500);
  text-transform: uppercase;
  letter-spacing: 0.07em;
  font-weight: 550;
  cursor: pointer;
  box-shadow: 0.1em 0.1em 0.2em var(--clr-neutral-900);

  &:hover,
  &:focus {
    background-color: var(--clr-primary-300);
  }

  &:focus {
    outline: none;
  }
}
