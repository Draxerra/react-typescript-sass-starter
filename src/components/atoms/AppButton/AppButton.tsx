import { FC } from "react";
import classNames from "classnames";

import styles from "./AppButton.module.scss";

type AppButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement>;

const AppButton: FC<AppButtonProps> = ({ children, className, ...props }) => {
  return (
    <button {...props} className={classNames(className, styles["app-button"])}>
      {children}
    </button>
  );
};

export default AppButton;
