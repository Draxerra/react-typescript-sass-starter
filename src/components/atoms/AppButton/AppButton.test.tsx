import AppButton from "./AppButton";
import { render } from "@testing-library/react";

describe("App Button", () => {
  it("should render", () => {
    const button = render(
      <AppButton className="test">Test</AppButton>
    ).asFragment();
    expect(button).toMatchSnapshot();
  });
});
