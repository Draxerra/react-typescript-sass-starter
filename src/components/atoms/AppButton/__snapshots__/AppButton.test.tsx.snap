// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Button should render 1`] = `
<DocumentFragment>
  <button
    class="test app-button"
  >
    Test
  </button>
</DocumentFragment>
`;
